lib/ccflay.rb

```diff
@@ -84,3 +84,25 @@ def pure_ruby_hash # :nodoc: see above
     hash
   end
 end
+
+class Sexp
+  def flatter
+    r = dup.clear
+
+    each do |s|
+      if s.is_a? Sexp
+        ss = s.flatter
+
+        if ss.size == 2 && ss[1].is_a?(Sexp)
+          r.concat ss
+        else
+          r << ss
+        end
+      else
+        r << s
+      end
+    end
+
+    r
```

Comment:
I'm having a hard time believing it costs that much... I'd like to understand that better, but maybe not on your dime.